### PR TITLE
Populate user-agent in HTTP clients.

### PR DESCRIPTION
**Pull request by anuraaga**
`http.user_agent` is a semantic attribute so we should fill it where we can.

### CODE REVIEW

**Comment by anuraaga** [comment]
Should have marked this as WIP - TBH running tests for a wide reaching PR like this one takes so long I can't really iterate locally.

I noticed, perhaps because of `buildSrc` not sure though, when running `test` it always runs everything without being up to date. We need to make sure the build is as incremental as possible before build cache will help us.

**Comment by iNikem** [comment]
Running 2 `./gradlew build` in a row works nicely for me: second build has almost all tasks up-to-date

**Comment by iNikem** [comment]
@anuraaga you still convert your PR into draft if you wish

**Comment by trask** [comment]
> @anuraaga you still convert your PR into draft if you wish

Oh! I've looked for the ability to do that before, but couldn't find. Had to google just now and found:

image

**Comment by anuraaga** [comment]
@iNikem Will try to understand where I see the non-incremental builds, indeed it doesn't seem to always be the case. But just noticed it after running tests after an amend commit with no changes at all (to kick circleci) - suspected the act of amend commit might affect the build state, but trying again that didn't seem to be the case. I see it happen though and will keep an eye out :)

**Comment by iNikem** [comment]
@anuraaga I have converted this PR to draft, as your "WIP" prefix suggests :) Please let me know if this was not your intention

**Comment by anuraaga** [comment]
Oops - I didn't read past @trask's "couldn't find" 🤣

**Comment by anuraaga** [comment]
Finally ready, sorry for the CI tag :)

File: instrumentation/khttp-0.1/src/main/java/io/opentelemetry/auto/instrumentation/khttp/KHttpDecorator.java [hunk 1]
```diff
@@ -42,4 +42,10 @@ protected URI url(RequestWrapper requestWrapper) throws URISyntaxException {
   protected Integer status(Response response) {
     return response.getStatusCode();
   }
+
+  @Override
+  protected String userAgent(RequestWrapper requestWrapper) {
+    // TODO: Find out how to get user agent.
```
Comment (iNikem) [inline instrumentation/khttp-0.1/src/main/java/io/opentelemetry/auto/instrumentation/khttp/KHttpDecorator.java:48]: File an issue about that?

Comment (anuraaga) [reply instrumentation/khttp-0.1/src/main/java/io/opentelemetry/auto/instrumentation/khttp/KHttpDecorator.java:48]: Thanks for the diligence! Filed https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/557

**Pull request merged**